daml-foundations/daml-ghc/tests/FunRetTypeSig.daml

```diff
@@ -7,4 +7,19 @@ module FunRetTypeSig where
 
 succ (x : Int) : Int = x + 1
 
--- Related issue : https://github.com/digital-asset/daml/issues/747.
+-- Regression tests for https://github.com/digital-asset/daml/issues/747
+
+-- Monomorphic test
+fact (n : Int) : Int
+  | n <= 1    = 1
+  | otherwise = n * fact (n - 1)
+
+-- Polymorphic test
+mymap (f : a -> b) (xs : [a]) : [b] = case xs of
+    [] -> []
+    x::xs -> f x::mymap f xs
+
+-- Test that the rewriting in the compiler works.
+testFact = scenario do
+  assert $ fact 5 == 120
```

Comment:
Can we test `succ 12 == 13` which we're at it?

Reply:
I'd personally do `fact 5 === 120` (`===` from `DA.Assert`) since then if they don't match we get a really helpful error message. I find `assert $ a == b` to be an anti-pattern.

Reply:
Done.